src/openrct2-ui/input/MouseInput.cpp

```diff
@@ -0,0 +1,1551 @@
+#pragma region Copyright (c) 2014-2017 OpenRCT2 Developers
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#include <openrct2/audio/audio.h>
+#include <openrct2/config/Config.h>
+#include <openrct2/Context.h>
+#include <openrct2/Game.h>
+#include <openrct2/Input.h>
+#include <openrct2/interface/chat.h>
+#include <openrct2/interface/console.h>
+#include <openrct2/interface/Cursors.h>
+#include <openrct2/interface/viewport.h>
+#include <openrct2/interface/widget.h>
+#include <openrct2/interface/window.h>
+#include <openrct2/localisation/localisation.h>
+#include <openrct2/platform/platform.h>
+#include <openrct2/ride/ride_data.h>
+#include <openrct2/scenario/scenario.h>
+#include <openrct2-ui/interface/Dropdown.h>
+#include <openrct2/world/banner.h>
+#include <openrct2/world/map.h>
+#include <openrct2/world/sprite.h>
+#include <openrct2/world/scenery.h>
+#include <openrct2/OpenRCT2.h>
+#include <openrct2/core/Math.hpp>
+#include <openrct2/core/Util.hpp>
+#include <openrct2-ui/windows/Window.h>
+
+typedef struct rct_mouse_data {
```

Comment:
Please reformat this file.

Reply:
I recall someone using the excuse "This deliberately skips reformatting, in order to get stuff to C++ quicker." elsewhere… :wink:

Reply:
Those are not refactor PRs. This one is.

Reply:
This isn't a refactor PR. It just moves stuff.

Reply:
How is that not refactoring? You're moving it to split UI and engine bits.
Anyway, I'm not going obstruct the PR for it, but I'd be disappointed if you didn't do at least an autoformat to correct braces and spaces.